### ISSUE TITLE
Questions about extracting passes from LLVM

### ISSUE THREAD

**Issue by chenyangyc** [ise-uiuc/WhiteFox#2]
Such fascinating work! I do have a minor inquiry: How might I access all passes within LLVM? I noticed in the paper that 52 optimizations and their implementations were extracted, yet the ./llvm-exec/example.json file only contains information for 20 optimizations. Additionally, I seem to be unable to locate any extraction scripts. Might there be something I've overlooked?

**Comment by YangChenyuan** [comment]
Thanks for your interest!

I uploaded all our collected LLVM optimizations in PR #3, including 52 optimizations.

Basically, we tried our best to manually collect the general-purpose LLVM optimization (not specific to the platform) from this documentation (https://llvm.org/docs/Passes.html, Transform passes). However, we failed to collect for several optimizations.

> unable to locate any extraction scripts

Yes, the optimization collection part is done manually and there aren't any automatic scripts.
